network/local.go

```diff
@@ -0,0 +1,459 @@
+package network
+
+import (
+	"errors"
+	"fmt"
+	"golang.org/x/net/context"
+	"reflect"
+	"sync"
+	"time"
+)
+
+// NewLocalRouter returns a fresh router which uses local connections. It takes
+// the default context.
+func NewLocalRouter(sid *ServerIdentity) (*Router, error) {
+	return NewLocalRouterWithContext(defaultLocalContext, sid)
+}
+
+// NewLocalRouterWithContext is the same as NewLocalRouter but takes a specific
+// LocalContext. This is useful to run parallel different local overlays.
```

Comment:
Corrected English. IIRC, `each` takes singular and is 3rd person, so `listen_s_` and `connect_s_`, and `each test` without s. But `other` takes plural, so `other listener_s_`. Any native English speakers?